### ISSUE TITLE
Add the matrix logrithm method logm() to Qobj.

### ISSUE THREAD

**Issue by DS-Liu** [qutip/qutip#1990]
### Problem Description

I don't find matrix logrithm logm() in Qobj methods, so to compute matrix logrithm I call logm() in scipy.linalg, which changes the type from qobj to ndarray. And to continue the subsequent computation, I need to reinstantiate the resulting ndarray to Qobj, which is cumbersome.
It would be better if the matrix logrithm logm() is added to the Qobj

### Proposed Solution

Add the matrix logrithm method logm() to Qobj.

**Comment by Ericgig** [comment]
It will be added it the next major release.

**Comment by hodgestar** [comment]
This won't be implemented in QuTiP 4.7, but one can implement it oneself for a given `qobj` using:

```python
        import scipy.linalg

        qobj_logm = Qobj(
            scipy.linalg.logm(qobj.full()),
            dims=self.dims,
            type=self.type
        )
```
